jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/quicknewjob/NewJobPresenterTest.java

```diff
@@ -0,0 +1,229 @@
+        when(executorServicesMock.scheduleRequest(eq(serverTemplateId),
+                                                  eq(JOB_TYPE),
+                                                  eq(dueDate),
+                                                  any())).thenReturn(jobId);
+        presenter.createJob(JOB_NAME,
+                            dueDate,
+                            JOB_TYPE,
+                            JOB_RETRIES,
+                            new ArrayList<RequestParameterSummary>());
+
+        verify(requestChangedEvent).fire(any(RequestChangedEvent.class));
+        verify(view).hide();
+
+        final ArgumentCaptor<NotificationEvent> captor = ArgumentCaptor.forClass(NotificationEvent.class);
+        verify(notificationEvent).fire(captor.capture());
+        assertEquals(1,
+                     captor.getAllValues().size());
+        assertEquals(NotificationEvent.NotificationType.DEFAULT,
+                     captor.getValue().getType());
+        assertEquals(Constants.INSTANCE.RequestScheduled(jobId),
+                     captor.getValue().getNotification());
+    }
+
+    @Test
+    public void testScheduleRequestParameters() {
+        String param_key = "param_key";
+        String param_value = "param_value";
+        Date dueOnDate = new Date();
+        final StringBuffer assertionsResults = new StringBuffer();
+        doAnswer(new Answer() {
+            @Override
+            public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
+                assertTrue(invocationOnMock.getArguments()[0].equals(serverTemplateId));
+                assertTrue(invocationOnMock.getArguments()[1].equals(JOB_TYPE));
+                assertTrue(invocationOnMock.getArguments()[2].equals(dueOnDate));
+                final HashMap<String, String> ctxValues = (HashMap) invocationOnMock.getArguments()[3];
+                assertTrue(ctxValues.get(RequestDataSetConstants.COLUMN_BUSINESSKEY).equals(JOB_NAME));
+                assertTrue(ctxValues.get(RequestDataSetConstants.COLUMN_RETRIES).equals(String.valueOf(JOB_RETRIES)));
+                assertTrue(ctxValues.get(param_key).equals(String.valueOf(param_value)));
+                assertionsResults.append("Success");
+                return null;
+            }
+        }).when(executorServicesMock).scheduleRequest(anyString(),
+                                                      anyString(),
+                                                      any(Date.class),
+                                                      any(Map.class));
+
+        presenter.createJob(JOB_NAME,
+                            dueOnDate,
+                            JOB_TYPE,
+                            JOB_RETRIES,
+                            Arrays.asList(new RequestParameterSummary(param_key,
+                                                                      param_value)));
+        assertEquals("Success",
+                     assertionsResults.toString());
```

Comment:
@nmirasch This approach definitely does the trick, but really looks more like a workaround rather than something we would like to repeat again somewhere else.
Also setting aside the look, the assert error is very scarce and gives no information which assertion failed and for which reason.
I found another, more elegant approach which I think you will like too. The solution relies on _AssertJ_ library, so you have to add to _jbpm-wb-executor-service-client/pom.xml_ the following dependency:
```
    <dependency>
      <groupId>org.assertj</groupId>
      <artifactId>assertj-core</artifactId>
      <scope>test</scope>
    </dependency>
```
Once you did that, you just have to replace a couple of things in your test:
1. ```final StringBuffer assertionsResults = new StringBuffer();``` 
with ```final SoftAssertions softly = new SoftAssertions();```
2. Each assertion statement ```assertTrue(invocationOnMock.getArguments()[1].equals(JOB_TYPE));```
with ```softly.assertThat(invocationOnMock.getArguments()[1]).isEqualTo(JOB_TYPE);```
3. 
```
assertEquals("Success",
             assertionsResults.toString());
```
with ```softly.assertAll();```

Now, you have a full description of a failure and even more better - if there are multiple assertion failures, all of them will be collected and presented :)